File: 83-remove-duplicates-from-sorted-list/remove-duplicates-from-sorted-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode* prev = head;
        ListNode* curr = head;
        while(curr!=NULL){
            if(curr->val!=prev->val){
                prev->next = curr;
                prev = curr;
            }
            curr = curr->next;
        }
        if(prev!=NULL)
            prev->next = NULL;
        return head;
    }
};

File: 83-remove-duplicates-from-sorted-list/remove-duplicates-from-sorted-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode slow=head;
        ListNode fast=head;
        while(fast!=null){
            if(slow.val!=fast.val){
                slow.next=fast;
                slow=fast;
            }
            fast=fast.next;
        }
        if(slow!=null)
        slow.next=null;
        return head;
    }
}